forge/db/models/Notification.js

```diff
@@ -89,6 +89,23 @@ module.exports = {
                         count,
                         notifications: rows
                     }
+                },
+                markNotificationsAsRead: async ({ read = true, ids = [] }, user) => {
+                    if (ids.length === 0) {
+                        return
+                    }
+
+                    ids = ids.map(id => Number.isNaN(id) ? M.Notification.decodeHashid(id) : id)
+
+                    await M.Notification.update(
+                        { read: read ? 1 : 0 },
+                        {
+                            where: {
+                                id: ids.map(M.Notification.decodeHashid),
```

Comment:
why are you performing decode twice? (line 98 and line 104)

Also, not sure why the `Number.isNaN` is being used.  Perhaps you meant `ids = ids.map(id => typeof id === 'string' ? M.Notification.decodeHashid(id) : id)`

Lastly, as the array of data could be "bad ID strings", have you checked the consequences of the resulting `id` array on the datebase?

e.g. if `id` = `['goodId1', 'goodId2', 'badId1', 'badId2', 'goodId3']` what does the resulting array `id` hold and what affect does it have on the database?  This is something i would hope to see in a unit test.